pvmismatch/pvmismatch_lib/pvsystem.py

```diff
@@ -81,13 +81,17 @@ def Istring(self):
     def Vstring(self):
         return np.asarray([pvstr.Vstring.flatten() for pvstr in self.pvstrs])
 
+    @property
+    def Voc_str(self):
+        return np.asarray([pvstr.Voc_mod.sum() for pvstr in self.pvstrs])
+
     def calcSystem(self):
         """
         Calculate system I-V curves.
         Returns (Isys, Vsys, Psys) : tuple of numpy.ndarray of float
         """
         Isys, Vsys = self.pvconst.calcParallel(
-            self.Istring, self.Vstring, self.Vstring.max(), self.Vstring.min()
+            self.Istring, self.Vstring, self.Voc_str.max(), self.Vstring.min()

```

Comment:
What do you think about this, that the system is bounded by the largest Voc, and not necessarily into the 4th quadrant? I figure why would you ever need negative current at the system level? No inverter would ever even get close to this right?

Reply:
Makes sense to me. So far it never occurred to me to instigate negative current on a system level. Also blocking diodes or a charge controller wouldn't really allow this anyway.